packages/flutter/lib/src/material/navigation_drawer.dart

```diff
@@ -347,14 +364,14 @@ class _NavigationDestinationBuilder extends StatelessWidget {
                   width: (navigationDrawerTheme.indicatorSize ?? defaults.indicatorSize!).width,
                   height: (navigationDrawerTheme.indicatorSize ?? defaults.indicatorSize!).height,
                 ),
-                Row(
-                  children: <Widget>[
-                    const SizedBox(width: 16),
-                    buildIcon(context),
-                    const SizedBox(width: 12),
-                    buildLabel(context),
-                  ],
-                ),
+
+                if (enabled)
+                  destinationBody
+                else
+                  Opacity(
```

Comment:
We should support MaterialState.disabled and resolve the default iconTheme and labelTextStyle `MaterialStateProperty<Color?>` theme values against the additional state (see _NavigationDrawerDefaultsM3).

That way developers can override the default appearance of disabled destinations.

Reply:
Thanks for the tip, will do the fixes tonight